pkg/segment/writer/segstore.go

```diff
@@ -1440,8 +1434,26 @@ Encoding Scheme for all columns single file
 func (ss *SegStore) FlushSegStats() error {
 
 	if len(ss.AllSst) <= 0 {
-		log.Errorf("FlushSegStats: no segstats to flush")
-		return errors.New("FlushSegStats: no segstats to flush")
+		found := 0
+		tsKey := config.GetTimeStampKey()
+		for cname, cwip := range ss.wipBlock.colWips {
+			if cwip.cbufidx > 0 {
+				log.Infof("FlushSegStats: sst nil but cname: %v, cwip.cbufidx: %v", cname,
+					cwip.cbufidx)
+				if cname != tsKey {
+					found += 1
+				}
+			}
+		}
+		// Flush is called only one of the cbufidx is >0, but if we find no columns
+		// with cbufidx > 0 other than timestamp column, only then declare this as an error
+		// else we won;t create a sst file
```

Comment:
typos:
```suggestion
		// Flush is called once one of the cbufidx is >0, but if we find no columns
		// with cbufidx > 0 other than timestamp column, only then declare this as an error
		// else we won't create a sst file
```

Reply:
nit: it might be better to move this comment above the `for` loop, since this comment explains why we're doing the loop to find non-timestamp columns

Reply:
addressed in the next PR